code/clue.js

```diff
@@ -48,12 +278,59 @@ const pickKiller = () => {
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerAge').innerHTML = ` Age:${mystery.killer.age}`
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+  let favWeapon = weapons.find(weapon => weapon.id === mystery.killer.favouriteWeapon)
+  document.getElementById('killerFavouriteWeapon').innerHTML = `Favourite weapon: ${favWeapon.name}`
+  document.getElementById('killerDescription').innerHTML = `Description: ${mystery.killer.description}`
+}
+document.getElementById('killerCard').onclick = pickKiller
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+
+
+  document.getElementById('weaponCard').style.background = mystery.weapon.color
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`
+  document.getElementById('weaponMortality').innerHTML = `Mortality: ${mystery.weapon.mortality}`
+
+}
+
+document.getElementById('weaponCard').onclick = pickWeapon
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+
+  document.getElementById('roomName').innerHTML = `${mystery.room.name}`
+  document.getElementById('roomCard').style.background = mystery.room.color
+
+
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+document.getElementById('roomCard').onclick = pickRoom
+
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+const revealMystery = () => {
+  if (mystery.killer === '' || mystery.weapon === '' || mystery.room === '') {
+  document.getElementById('mystery').innerHTML = 'Sorry, you have to pick one of each card to reveal the murder mystery'
+  } else {
+  document.getElementById('mystery').innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName} in the ${mystery.room.name} with a ${mystery.weapon.name}.`
+  }
+}
+
+document.getElementById('revealMystery').onclick = revealMystery
+
+  const shuffleFavouriteWeapon = () => {
+    suspects.forEach((suspect) => {
```

Comment:
No need for extra pair of parentheses around `suspect` if you only have one argument. It can (and even should) be written like this :
```
suspects.forEach(suspect => { ... }
```